### PR TITLE
Update IGCE .xlsx Template

### PR DESCRIPTION
**Pull request by zachclark-ccpo**
# Overview
Uses the template linked here: https://docs.google.com/spreadsheets/d/1JebJPfTZI_9LJxLMgVU51tmdrAqdSgTL/edit#gid=1682641462

Complete refactor of IGCE generation, now most formulas for the Period sheets are handled in typescript.

### Summary sheet
image
Summary sheet now includes dropdowns/ single line calculations for selected CLIN
image
The CLINs selected for the Summary sheet will dynamically update based on which CLIN is selected.
For example: Changing the CLIN to **x007 Travel** will dynamically recalculate the sheet:
image
This behavior was accomplished by writing a new excel formula:
`=IFERROR(VLOOKUP(A6,'Base Period'!A41:I54,9, FALSE),0)`
The table array: `A41:I54,`, is dynamically generated and accounts for any additional rows that are added. 

### Base Period sheet with lots of items
image
image
### Option Period 1 with few items
image

Ticket:AT-8383

### CODE REVIEW

**Review by juliusfitzhugh-ccpo — APPROVED** [review]
Nice improvements in the generation of the document! Looks good, I didn't see anything standing out.

File: document-generation/igce-document.ts [hunk 1]
```diff
@@ -45,239 +42,225 @@ export async function generateIGCEDocument(
...
-  let summarySheetIdiqClinCounter = 0; // keep count of unique idiq clins of each period
 
-  // function that populates each period sheet
+  /* Period sheet helper is used to get the specific cell reference ranges for the summary sheet
+    that are generated during populatePeriodLineItems.
+    Example output for periodSheetHelper: 
+      { period: 'Base Period', cellRef: 'A41:I54' }
+      { period: 'Option Period 1', cellRef: 'A28:I41' }
+      { period: 'Option Period 2', cellRef: 'A28:I41' }
+  */
```
Comment (jasonburkert-ccpo) [inline document-generation/igce-document.ts:77]: Love these explainers

File: document-generation/igce-document.ts [hunk 2]
```diff
@@ -45,239 +42,225 @@ export async function generateIGCEDocument(
...
+  const periodSheetHelper: Array<{ period: string; cellRef: string }> = [];
+
+  const headers = {
+    "Content-Type": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+    "Content-Disposition": `attachment; filename=IndependentGovernmentCostEstimate.xlsx`,
```
Comment (jasonburkert-ccpo) [inline document-generation/igce-document.ts:82]: Thinking out loud: Does EDMS care about specific file names?

**Review by jasonburkert-ccpo — APPROVED** [review]

**Comment by jasonburkert-ccpo** [comment]
Nice work on the generation code and the template.

Minor nitpick: On the template, several tabs, column C has a header **Contract Type**, but due to col width only  **Contract** is visible.

**Pull request merged**